app/browser/menu.js

```diff
@@ -83,7 +83,12 @@ const createFileSubmenu = () => {
       label: locale.translation('closeTab'),
       accelerator: 'CmdOrCtrl+W',
       click: function () {
-        appActions.tabCloseRequested(tabState.TAB_ID_ACTIVE)
+        const tabId = tabState.TAB_ID_ACTIVE
+        // Do not allow closing a pinned tab.
+        const isPinned = tabState.isTabPinned(state, tabId)
+        if (!isPinned) {
+          appActions.tabCloseRequested(tabId)
+        }
```

Comment:
This might also prevent closing through the File menu. I don't think that's desired.

is there a way to see if this was invoked via the keyboard accelerator or via the file menu?